packages/discovery/src/discovery/output/saveDiscoveryResult.ts

```diff
@@ -63,6 +102,141 @@ export async function saveDiscoveryResult(
   }
 }
 
+async function saveFlatSources(
+  rootPath: string,
+  results: Analysis[],
+  logger: DiscoveryLogger,
+  options: {
+    rootFolder?: string
+    sourcesFolder?: string
+    discoveryFilename?: string
+  },
+): Promise<void> {
+  const flatSourcesFolder = `.flat${options.sourcesFolder ?? ''}`
+  const flatSourcesPath = posix.join(rootPath, flatSourcesFolder)
+  const allContractNames = results.map((c) =>
+    c.type !== 'EOA' ? c.name : 'EOA',
+  )
+
+  await rimraf(flatSourcesPath)
+
+  logger.log(`Saving flattened sources`)
+  for (const contract of results) {
+    try {
+      if (contract.type === 'EOA') {
+        continue
+      }
+
+      for (const [sourceIndex, source] of contract.source.entries()) {
+        const input: FileContent[] = Object.entries(source.files)
```

Comment:
Good catch, it's already fixed. It will skip this source, logging that it happened and informing the user that it's because we found zero `.sol` files.